On boarding flow

LinnierGames opened the pull request:
### What has Changed
- The user is logged in __and persisted__ **after** the user is taken from the login flow to the home tab bar

### What's New
- Logout from the SettingsVc
- OnBoardingDirector manages what view controller should be the next Vc depending on what the user has already accepted (location, verify phone number)

### What's Missing
> This checklist has been moved to [this](https://trello.com/c/KKaRgWfp) trello card
- [ ] Populate the images in the tutorial
- [ ] Shorten the descriptions of the tutorial

LinnierGames committed:
Create OnBoardingDistributer

This will return the view controller to be presented by the caller

LinnierGames committed:
Use OnBoardingDistributer

LinnierGames committed:
WIP: Present additional Vcs

TODO: hide back button

LinnierGames committed:
Docs

LinnierGames committed:
WIP: Logout

Using BasicSettingsItem, invoke the logout in UserService

TODO: dismiss/present the FrontPageVc

LinnierGames committed:
WIP: Present/Dismiss to/from Front Page/HomeVc

LinnierGames committed:
Display/Dismiss OnBoarding Screesn

LinnierGames committed:
Add `hasSeenTutorial`

TODO: Present the TutorialVc

LinnierGames committed:
Present Tutorial

LinnierGames committed:
Fix Missing User Property

LinnierGames committed:
Docs

LinnierGames committed:
Open the Settings

If the user has denied to allow location services, prompt the user to open the settings to enable location. The buttons are updated when the app returns from being in the background

LinnierGames committed:
docs

LinnierGames committed:
WIP: Tutorial Content

Needs to be shorter

LinnierGames committed:
Merge branch 'dev' into erick-on-boarding

LinnierGames commented:
![](https://github.trello.services/images/mini-trello-icon.png) [On-boarding Flow](https://trello.com/c/4yjGmfy1/31-on-boarding-flow)

pondorasti commented on KifuSF/KFCLocationServiceDisclaimer.swift:
```diff
@@ -62,13 +107,18 @@ extension KFCLocationServiceDisclaimer: UIConfigurable {
         
         title = "Location Privacy"
-        locationServiceDisclaimerLabel.text = "In order to use Kifu we will need to know you location only while using the app."
+        locationServiceDisclaimerLabel.text = """
```
This could be handled in the configureData method of UIConfigurable.

Pondorasti commented:
I have noticed that there are some TODOs in KFCPhoneNumberValidation, what is the status of them? are they supposed to be finished in this PR?

pondorasti reviewed (approved):
Everything looks good! Maybe the TODOs could be fixed in a latter PR.

LinnierGames committed:
Update changes to configureData

LinnierGames committed:
Remove TODOs

TODO: Change the noCodeLabel to a button

LinnierGames committed:
Throttle the Send Text Message

LinnierGames commented:
@Pondorasti i've added the changes you've mentioned. If i could use your help with change the uilabel to a button for the noCodeLabel that would be great! I've added a `sendTextMessage()` method [here](https://github.com/Kifuapp/KifuSF/pull/35/files#diff-5473e2e0e26eea21c9ceef0e8d9d8913R91) to make it easier when setting up the button.

LinnierGames committed:
Merge branch 'dev' into erick-on-boarding

LinnierGames commented:
@Pondorasti , @woodward4422 will be finishing the rest of the checkboxes in the RP description before I merge this in

LinnierGames committed:
Merge remote-tracking branch 'origin/dev' into erick-on-boarding

LinnierGames committed:
Merge remote-tracking branch 'refs/remotes/origin/erick-on-boarding' into erick-on-boarding
